lib/SIL/Utils/BasicBlockUtils.cpp

```diff
@@ -465,25 +477,28 @@ void JointPostDominanceSetComputer::findJointPostDominatingSet(
     // our initial blocks (signaling a loop) and then add it to the worklist if
     // we haven't visited it already.
     for (auto *predBlock : block->getPredecessorBlocks()) {
-      if (initialBlocks.count(predBlock)) {
+      if (initialBlocks.contains(predBlock)) {
         reachableInputBlocks.push_back(predBlock);
         for (auto *succBlock : predBlock->getSuccessorBlocks()) {
-          if (visitedBlocks.count(succBlock))
+          if (visitedBlocks.contains(succBlock))
             continue;
           if (deadEndBlocks.isDeadEnd(succBlock))
             continue;
-          blocksThatLeakIfNeverVisited.insert(succBlock);
+          blocksThatLeakIfNeverVisited.push_back(succBlock);
+          isLeakingBlock.set(succBlock);
         }
       }
-      if (visitedBlocks.insert(predBlock).second)
+      if (visitedBlocks.insert(predBlock))
         worklist.push_back(predBlock);
     }
   }
 
   // After our worklist has emptied, any blocks left in
   // blocksThatLeakIfNeverVisited are "leaking blocks".
-  for (auto *leakingBlock : blocksThatLeakIfNeverVisited)
-    foundJointPostDomSetCompletionBlocks(leakingBlock);
+  for (auto *leakingBlock : blocksThatLeakIfNeverVisited) {
```

Comment:
I don't think this is sufficient. How do you handle if we have the same successor block that must be visited for two different blocks. Then we would call foundJointPostDomSetCompletion with the same block twice which breaks an API guarantee.

Reply:
you are right, good catch!
I'll fix it.

Reply:
Can you fix the comment here? I think the comment needs to be updated given functional changes.